pkg/destinations/redshift/insert.go

```diff
@@ -0,0 +1,133 @@
+package redshift
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/ilyakaznacheev/cleanenv"
+	"github.com/rs/zerolog/log"
+	"github.com/scratchdata/scratchdata/config"
+	"github.com/scratchdata/scratchdata/pkg/storage/blobstore/s3"
+	"github.com/scratchdata/scratchdata/util"
+)
+
+func (s *RedshiftServer) createColumns(table string, jsonTypes map[string]string) error {
+	for colName, jsonType := range jsonTypes {
+		colType := "VARCHAR"
+		switch jsonType {
+		case "int":
+			colType = "BIGINT"
+		case "bool":
+			colType = "BOOLEAN"
+		case "float":
+			colType = "DOUBLE PRECISION"
+		case "string":
+			colType = "VARCHAR"
+		default:
+			colType = "VARCHAR"
+		}
+
+		sql := fmt.Sprintf("ALTER TABLE \"%s\" ADD COLUMN IF NOT EXISTS \"%s\" %s", table, colName, colType)
+		_, err := s.conn.Exec(sql)
+		if err != nil {
+			return err
+		}
+
+		return nil
+
+	}
+
+	return nil
+}
+func (s *RedshiftServer) CreateColumns(table string, fileName string) error {
+
+	input, err := os.Open(fileName)
+	if err != nil {
+		return err
+	}
+	// Infer JSON types for the input
+	jsonTypes, err := util.GetJSONTypes(input)
+	if err != nil {
+		return err
+	}
+
+	err = s.createColumns(table, jsonTypes)
+	if err != nil {
+		return err
+	}
+
+	err = input.Close()
+	if err != nil {
+		log.Error().Err(err).Str("filename", fileName).Msg("Unable to close file")
+	}
+
+	return nil
+
+}
+
+func (s *RedshiftServer) CreateEmptyTable(table string) error {
+	sql := fmt.Sprintf("CREATE TABLE IF NOT EXISTS \"%s\" (__row_id BIGINT)", table)
+	_, err := s.conn.Exec(sql)
+	return err
+}
+
+func (s *RedshiftServer) InsertFromNDJsonFile(table string, fileName string) error {
+	var configOptions config.ScratchDataConfig
+	err := cleanenv.ReadConfig(os.Args[1], &configOptions)
```

Comment:
Instead of reading the S3 configuration options from a file, we should be using the values from the `RedshiftConfiguration` struct. Every Redshift configuration will have a different S3 config.